### ISSUE TITLE
Can't proxy request when target server is HTTPS

### ISSUE THREAD

**Issue by OliverJAsh** [typicode/hotel#160]
Hi,

I have a server running at `https://localhost:9000`. I added this as an alias in hotel:

```
❯ cat ~/.hotel/servers/cubik.json
{
  "target": "https://localhost:9000"
}
```

However, when I try to access the alias `cubik.dev`, I get this response:

```
Can't proxy request to https://localhost:9000
```

Accessing https://localhost:9000 works fine.

I tried with a different server on HTTP and that seemed to work OK, so I'm not sure what the problem could be.

Here is my daemon log, if that helps:

```
10 Jan 09:11:51 - Resolve cubik
10 Jan 09:11:51 - Proxy http://cubik.dev to https://localhost:9000
10 Jan 09:11:51 - Proxy error
10 Jan 09:11:52 - Resolve cubik
10 Jan 09:11:52 - Proxy http://cubik.dev to https://localhost:9000
10 Jan 09:11:52 - Proxy error
```

**Comment by typicode** [comment]
Hi @OliverJAsh,

There seem to be an issue when target is `https`. If you try proxying to `http://localhost:9000`, do you have the issue? You can still go https://cubik.dev.

**Comment by OliverJAsh** [comment]
Thanks @typicode. That appears to be the issue. I'll update the title to reflect that.

**Comment by typicode** [comment]
Thank you.
I'll have a look, I'm almost sure it worked before so maybe it's related to a recent update.

**Comment by typicode** [comment]
When the request is proxied, `host` is `cubik.dev` and it may not match the hosts from your certificate.
If possible, I would create a dev certificate with `cubik.dev` in it and see if it works.

Next version of `hotel` will show and log proxy errors to make debugging easier.

**Comment by typicode** [comment]
Hi @OliverJAsh,

Just released `v0.6.0`, you can now pass `--co/--change-origin` when adding a remote server. It will change the host header to match the remote host. This should solve the case where host doesn't match the one in the certificate.

```sh
hotel add --co https://cubik.dev
```

**Comment by OliverJAsh** [comment]
Thanks @typicode
